massa-protocol-worker/src/worker_operations_impl.rs

```diff
@@ -193,8 +198,7 @@ impl ProtocolWorker {
 
     /// Process the reception of a batch of asked operations, that means that
     /// we sent already a batch of ids in the network notifying that we already
-    /// have those operations. Ask pool for the operations (will change with
-    /// the shared storage)
+    /// have those operations. Ask pool for the operations.
     ///
     /// See also `on_operation_results_from_pool`
     pub(crate) async fn on_asked_operations_received(
```

Comment:
It's debatable, and I think it's better to ask pool, because then we can assert that the operations are in fact present in storage, otherwise we cannot(in protocol), see the difference between "not there but should be there(error)" and "not there because it's not supposed to be there".

Reply:
Because we're dealing with mpsc timeouts, I thought that we would like to reduce the communication between modules. That said, I understand your point... if we store the operations inside _pool_ instead of _protocol_, would we be sure that a load failed only in the case _"not there because it's not supposed to be there"_?  

About that, a distant node never ask another node something not declared with some batch ids.

So in a nominal case, we should never fail to load operations, right? And if a malicious (:smirk:)  ask us something that we don't have, both _"not there but should be"_ and _"not supposed to be here"_ result to the same behavior?

Reply:
Yes, I think it could be a nice follow-up, especially if we can somehow know for sure that "it doesn't matter" from a logical perspective.